src/piggui.rs

```diff
@@ -160,17 +162,28 @@ impl Application for Gpio {
                 self.chosen_layout = layout;
             }
             Message::ConfigLoaded((filename, config)) => {
-                self.config_filename = Some(filename);
+                self.config_filename = Some(filename.clone());
                 self.gpio_config = config.clone();
 
-                for (pin_number, pin_function) in config.configured_pins.iter() {
-                    self.pin_function_selected[*pin_number as usize - 1] = Some(*pin_function);
-                }
-
                 self.update_hw_config();
             }
-
             Message::None => {}
+            Message::Save => {
+                // Synchronous file handling
+                if let Some(path) = FileDialog::new().set_title("Choose files").save_file() {
+                    let path_str = path.to_str().unwrap();
+
+                    if let Err(err) = self.gpio_config.save(path_str) {
+                        eprintln!("Error saving configuration to {}: {}", path.display(), err);
```

Comment:
I added an issue to review and fix all our erro rhandling later, this is fine for now I think.